File: Examples/Python/Containers/MongoDB-REST/src/services/MongoDB/CRUD/mongodb_config.py
class MongoDB_Config:
    def __init__(self, username, password, server, retryWrites, writeConcern, appName, minPoolSize, maxPoolSize, waitQueueTimeoutMS):
        self.username = username
        self.password = password
        self.server = server
        self.retryWrites = retryWrites
        self.writeConcern = writeConcern
        self.appName = appName
        self.minPoolSize = minPoolSize
        self.maxPoolSize = maxPoolSize
        self.waitQueueTimeoutMS = waitQueueTimeoutMS

    def to_uri(self) -> str:
        return f"mongodb+srv://{self.username}:{self.password}@{self.server}/?retryWrites={self.retryWrites}&w={self.writeConcern}&appName={self.appName}&minPoolSize={self.minPoolSize}&maxPoolSize={self.maxPoolSize}&waitQueueTimeoutMS={self.waitQueueTimeoutMS}"

File: Examples/Rust/Containers/MongoDB-REST/src/src/services/mongodb/crud/mongodb_config.rs
pub struct MongoDBConfig {
    pub username: String,
    pub password: String,
    pub server: String,
    pub retry_writes: bool,
    pub write_concern: String,
    pub app_name: String,
    pub min_pool_size: u32,
    pub max_pool_size: u32,
    pub wait_queue_timeout_ms: u64,
}

impl MongoDBConfig {
    pub fn to_uri(&self) -> String {
        format!(
            "mongodb+srv://{}:{}@{}/?retryWrites={}&w={}&appName={}&minPoolSize={}&maxPoolSize={}&waitQueueTimeoutMS={}",
            self.username,
            self.password,
            self.server,
            self.retry_writes,
            self.write_concern,
            self.app_name,
            self.min_pool_size,
            self.max_pool_size,
            self.wait_queue_timeout_ms
        )
    }
}